cms/djangoapps/contentstore/management/commands/set_advanced_fields_all_courses.py

```diff
@@ -0,0 +1,248 @@
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Duraration:
+        When running this against all courses it may take considerable more time because
+        the course publishes out. This is due to the `update_course_advanced_settings` call.
+
+    Invoke with:
+
+        # Single Course
+        # Set the following advanced settings:
+        # `invitation_only` to `true`
+        # `catalog_visilibilty` to `about`
+        python manage.py cms --settings=devstack_docker set_advanced_fields_all_courses \
+        --fields_json '{"invitation_only": {"value": "true"}}' --user edx@example.com --course course-v1:ORG+COURSENUMBER+COURSERUN --force-update-catalog-visibility-about # pylint: disable=line-too-long
+
+        # All Courses
+        # Set the following advanced settings:
+        # `invitation_only` to `true`
+        # `catalog_visilibilty` to `about`
+        # `cerfificate_display_behavior` to `early_no_info`
+        python manage.py cms --settings=devstack_docker set_advanced_fields_all_courses \
+        --fields_json '{"invitation_only": {"value": "true"}, "certificates_display_behavior": {"value": "early_no_info"}}' --user edx@example.com --course course-v1:ORG+COURSENUMBER+COURSERUN --force-update-catalog-visibility-about # pylint: disable=line-too-long
+
+        # All Courses
+        # Set the following advanced settings:
+        # `catalog_visilibilty` to `about`
+        python manage.py cms --settings=devstack_docker set_advanced_fields_all_courses \
+        --fields_json '{}' --user edx@example.com --course course-v1:ORG+COURSENUMBER+COURSERUN --force-update-catalog-visibility-about # pylint: disable=line-too-long
+    """
+    help = 'Set an advanced settings for all courses based on a json list of fields.'
+
+    fields_json_help = '\
+        Need to have a json list of "field_name": "<new value>" \
+        { \
+            "catalog_visibility": { \
+                "value": "about" \
+            }, \
+            "certificates_display_behavior": { \
+                "value": "early_no_info" \
+            }, \
+            "invitation_only": { \
+                "value": "true" \
+            } \
+        }'
+    user_help = '--user <email> required, needs to be a staff member'
+    course_help = '--course <id> required, e.g. course-v1:ORG+COURSENUMBER+COURSERUN'
+    force_update_catalog_visibility_about_help = '--force-update-catalog-visibility-about, To fix `There was an error loading this course` error when loading MFE learning this forces `catalog_visibility` from `none` to `about`.'  # pylint: disable=line-too-long
+    limit_bulk_update_operation_help = '--limit-bulk-update-operation {0-N}'
+
+    _force_update_catalog_visibility_about = False
+    _updated_catalog_visilibity_about = []
+
+    def add_arguments(self, parser):
+        parser.add_argument('--fields_json',
+                            dest='fields_json',
```

Comment:
I think for consistency, the command-line parameter should use a hyphen: `--fields-json` not `--fields_json`